journal.py

```diff
@@ -1,27 +1,66 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
-import sqlalchemy as sa
-from sqlalchemy.ext.declarative import declarative_base
 import os
-import datetime
 from pyramid.config import Configurator
 from pyramid.view import view_config
 from waitress import serve
+import datetime
+import sqlalchemy as sa
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import scoped_session, sessionmaker
+from zope.sqlalchemy import ZopeTransactionExtension
+from pyramid.httpexceptions import HTTPNotFound
+#--maybe--
+from sqlalchemy.exc import IntegrityError
+#---------
 
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+Base = declarative_base()
+DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    'postgresql://jameshemmaplardh:123@localhost:5432/learning-journal'
+)
 
-# DATABASE_URL = os.environ.get(
-#     'DATABASE_URL',
-#     'postgresql://jameshemmaplardh:123@localhost:5432/learning-journal'
-# )
 
+class Entry(Base):
+    __tablename__ = 'entries'
+    
+    id = sa.Column(sa.Integer, primary_key=True)
+    title = sa.Column(sa.Unicode(127), nullable=False, unique=True)
+    text = sa.Column(sa.Unicode(), nullable=False)
+    created = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow)
 
+    @classmethod
+    def write(cls, title=None, text=None, session=None):
+        if title == "":
+            raise ValueError()
+        if not session:
+            session = DBSession
+        instance = cls(title=title, text=text)
+        session.add(instance)
+        return instance
+    
+    @classmethod
+    def all(cls, session=None):
+        if session == None:
```

Comment:
Better to write `if session is None` --- http://docs.quantifiedcode.com/python-code-patterns/readability/comparison_to_none.html